src/components/sbb-checkbox/sbb-checkbox.tsx

```diff
@@ -0,0 +1,138 @@
+import { Component, Prop, h, Event, EventEmitter, Watch, JSX } from '@stencil/core';
+import { InterfaceCheckboxAttributes } from './sbb-checkbox.custom';
+
+let nextId = 0;
+@Component({
+  shadow: true,
+  styleUrl: 'sbb-checkbox.scss',
+  tag: 'sbb-checkbox',
+})
+export class SbbCheckbox {
+  private _checkbox: HTMLInputElement;
+
+  /** Whether the checkbox is checked. */
+  @Prop({ mutable: true }) public checked: boolean;
+
+  /** Value of checkbox. */
+  @Prop() public value?: string;
+
+  /** Name of the checkbox */
+  @Prop() public name?: string;
+
+  /** Id of the internal input element - default id will be set automatically. */
+  @Prop() public inputId = `sbb-checkbox-${++nextId}`;
+
+  //** the svg for the true state */
+  @Prop() public labelIcon: '';
+
+  /** The disabled prop for the disabled state. */
+  @Prop() public disabled!: boolean;
+
+  /** The required prop for the required state. */
+  @Prop() public required?: boolean;
+
+  /** Whether the checkbox is tristated. */
+  @Prop() public tristated?: boolean;
+
+  /** The label position relative to the labelIcon. Defaults to false */
+  @Prop() public labelPosition: InterfaceCheckboxAttributes['labelPosition'] = 'after';
+
+  /** Whether the checkbox label has spacing to the labelIcon. */
+  @Prop() public labelSpace = false;
```

Comment:
I do wonder whether this even needs a property. This is only relevant when used in a checkbox group and when the icon-placement is end. In this scenario it might make more sense to control this via CSS property or part selector.